sunpy/map/maputils.py

```diff
@@ -124,15 +124,37 @@ def sample_at_coords(smap, coordinates):
     return smap.data[smap.wcs.world_to_array_index(coordinates)]
 
 
+def _convert_pixels(edge):
+    """
+    Helper function to convert a list of edge pixels of the form [(x0, y0), (x1, y1),..., (xn,yn)]
+    into the form ([x0, x1,...,xn], [y0, y1,...,yn])
+
+    Parameter
+    ---------
+    edge : list
+        A list of pairs of tuples.
+
+    Returns
+    -------
+    tuple
+        A tuple containing two lists. The first entry is a list containing the first coordinate of
+        each pixel, and similarly for the second entry.
+    """
+    x = [p[0] for p in edge.value] * u.pix
+    y = [p[1] for p in edge.value] * u.pix
+    return x, y
```

Comment:
This looks slow (performance-wise).  At the very least, the list comprehension should not be done twice.

Reply:
I could just replace `map_edges` with a function that returns the pixels in the form that they are needed.

Reply:
isn't this `zip(egde.value)`? or the reverse of that maybe?